### PR TITLE
[NPM] test if 600+ connections are handled correctly

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

Test if 600+ connections are sent in 2 followed payloads with a maximum of 600 connections per payload

### Motivation

Improve our NPM test E2E

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/tests/npm/ec2_1host_test.go [hunk 1]
```diff
@@ -92,6 +98,66 @@ func (v *ec2VMSuite) TestFakeIntakeNPM() {
...
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for n payloads and check if the last 2 have a maximum span of 100ms
+func (v *ec2VMSuite) TestFakeIntakeNPM_600pluscnx() {
+	t := v.T()
+
+	targetHostnameNetID := ""
+	// looking for 1 host to send CollectorConnections payload to the fakeintake
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate a connection
+		v.Env().RemoteHost.MustExecute("ab -n 600 -c 600 http://www.datadoghq.com/")
```
Comment (hmahmood) [inline test/new-e2e/tests/npm/ec2_1host_test.go:112]: Should this be https?

File: test/new-e2e/tests/npm/ec2_1host_test.go [hunk 2]
```diff
@@ -92,6 +98,66 @@ func (v *ec2VMSuite) TestFakeIntakeNPM() {
...
+
+		hostnameNetID, err := v.Env().FakeIntake.Client().GetConnectionsNames()
+		assert.NoError(c, err, "GetConnectionsNames() errors")
+		if !assert.NotEmpty(c, hostnameNetID, "no connections yet") {
```
Comment (hmahmood) [inline test/new-e2e/tests/npm/ec2_1host_test.go:116]: Should this test be `assert.Len(c, hostnameNetID, 1)`?

File: test/new-e2e/tests/npm/ec2_1host_test.go (hunk 1 above)
Comment (hmahmood) [reply test/new-e2e/tests/npm/ec2_1host_test.go:112]: Also, does this need to be in the Eventually loop? Or outside of it and only checking for the payload below needs to be inside the Eventually loop?

File: test/new-e2e/tests/npm/ec2_1host_test.go [hunk 3]
```diff
@@ -92,6 +98,66 @@ func (v *ec2VMSuite) TestFakeIntakeNPM() {
...
+		}
+
+		cnx.ForeachConnection(func(c *agentmodel.Connection, cc *agentmodel.CollectorConnections, hostname string) {
+			assert.LessOrEqualf(t, len(cc.Connections), 600, "too many payloads")
+		})
+
+		found600plusConnections := false
+		hostPayloads := cnx.GetPayloadsByName(targetHostnameNetID)
+		lenHostPayloads := len(hostPayloads)
+		for i, cc := range hostPayloads {
```
Comment (hmahmood) [inline test/new-e2e/tests/npm/ec2_1host_test.go:140]: Do we need this loop? Wouldn't checking `hostPayloads[lenHostPayloads-2].Connections` be enough?

File: test/new-e2e/tests/npm/ec2_1host_test.go (hunk 2 above)
Comment (nplanel) [reply test/new-e2e/tests/npm/ec2_1host_test.go:116]: only one agent is deployed at the moment so indeed len would be 1, but goal would be to reuse these tests later in different environment.

File: test/new-e2e/tests/npm/ec2_1host_test.go (hunk 3 above)
Comment (nplanel) [reply test/new-e2e/tests/npm/ec2_1host_test.go:140]: yes it's enough, I tried to do 2 things in the loop in previous iteration
thanks

File: test/new-e2e/tests/npm/ec2_1host_test.go [hunk 4]
```diff
@@ -92,6 +99,58 @@ func (v *ec2VMSuite) TestFakeIntakeNPM() {
...
 
+// TestFakeIntakeNPM_600cnx_bucket Validate the agent can communicate with the (fake) backend and send connections
+// every 30 seconds with a maximum of 600 connections per payloads, if more another payload will follow.
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for n payloads and check if the last 2 have a maximum span of 100ms
+func (v *ec2VMSuite) TestFakeIntakeNPM_600cnx_bucket() {
+	t := v.T()
+
+	// generate connections
+	v.Env().RemoteHost.MustExecute("ab -n 600 -c 600 http://www.datadoghq.com/")
```
[the remark this answers is not in the document]
Comment (nplanel) [reply test/new-e2e/tests/npm/ec2_1host_test.go:110]: I think it's fine to hit cloudflare and we known this endpoint should be up all the time
not sure deploying a http server doesn't add value to the tests

[the remark this answers is not in the document]
Comment (nplanel) [reply test/new-e2e/tests/npm/ec2_1host_test.go:110]: on another hand we will hit statistics on http:// does someone look at that ?

[the remark this answers is not in the document]
Comment (hmahmood) [reply test/new-e2e/tests/npm/ec2_1host_test.go:110]: Yes that and the increased load is what I had in mind. This running thousands of times a day in ci could skew the stats and load.

[the remark this answers is not in the document]
Comment (nplanel) [reply test/new-e2e/tests/npm/ec2_1host_test.go:110]: I will add and deploy an httpbin server on pulumi

[the remark this answers is not in the document]
Comment (nplanel) [reply test/new-e2e/tests/npm/ec2_1host_test.go]: done

**Review by hmahmood — APPROVED** [review]

File: test/new-e2e/tests/npm/dockercompose_httpbin.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+			},
+		}
+		data, err := yaml.Marshal(agentManifest)
+		return string(data), err
+	}).(pulumi.StringOutput)
+
+	return docker.ComposeInlineManifest{
+		Name:    "httpbin",
+		Content: httpbinManifestContent,
+	}
```
Comment (pducolin) [inline test/new-e2e/tests/npm/dockercompose_httpbin.go:30]: **💬 suggestion**
Should be possible to load a manifest using a file, embedding it with `embed`, and pass it to the docker manager

Comment (pducolin) [reply test/new-e2e/tests/npm/dockercompose_httpbin.go:30]: You can load it as a string and use the `pulumi.String` to convert it to the expectdd `Content` type

File: test/new-e2e/tests/npm/ec2_1host_test.go [hunk 6]
```diff
@@ -92,15 +154,76 @@ func (v *ec2VMSuite) TestFakeIntakeNPM() {
...
+		assert.NoError(c, err, "GetConnectionsNames() errors")
+		if !assert.NotEmpty(c, hostnameNetID, "no connections yet") {
+			return
+		}
+		targetHostnameNetID = hostnameNetID[0]
+
+		t.Logf("hostname+networkID %v seen connections", hostnameNetID)
+	}, 60*time.Second, time.Second, "no connections received")
+
+	debugString := ""
```
Comment (pducolin) [inline test/new-e2e/tests/npm/ec2_1host_test.go:181]: **❓ question**
Is this meant to be returned ? Looks like it is written to, but never read

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/tests/npm/ec2_1host_test.go (hunk 6 above)
Comment (nplanel) [reply test/new-e2e/tests/npm/ec2_1host_test.go:181]: thanks, I forgot to add the debugString to the Eventually failed message, but I will drop this debug log

File: test/new-e2e/tests/npm/dockercompose_httpbin.go (hunk 5 above)
Comment (nplanel) [reply test/new-e2e/tests/npm/dockercompose_httpbin.go]: @pducolin seems we need a top level mapping object
got the error `
            error: Process exited with status 15: running " docker-compose -f $HOME/httpbin-compose-tmp/docker-compose-httpbin.yml up --detach --wait --timeout 300":
            Top-level object must be a mapping`

Comment (nplanel) [reply test/new-e2e/tests/npm/dockercompose_httpbin.go]: seems related to the double quote issue https://stackoverflow.com/questions/46550348/docker-stack-deploy-error-about-top-level-object-mappings

**Pull request merged**